draft-ietf-quic-transport.md

```diff
@@ -4340,8 +4340,9 @@ The STREAM_ID_BLOCKED frame contains a single field.
 
 Stream ID:
 
-: A variable-length integer indicating the highest stream ID that the sender
-  was permitted to open.
+: A variable-length integer indicating the stream ID limit at the time the
+  sender was unable to open new streams.
```

Comment:
Same issue here.

I am afraid that the text might still imply the same thing; i.e. the maximum stream ID that the sender was permitted to open at the time the sender was unable to open the streams.

Reply:
Hmm, that's a tricky one, changing the definition of MAX_STREAM isn't that easy.

With stream IDs, if I receive MAX_STREAM_ID of `N`, then the test for whether something is allowed is `id <= N`.  With stream octets, the test is `offset < MAX_STREAM_DATA`.  That makes this harder.

So with the latter, saying that you can't send `offset` is clean and simple (wording notwithstanding).  With streams though, you would be saying "I can only send `N` (but want to send `N+4`)", which is awkward, I agree.

Is changing the definition of MAX_STREAM_ID worth it?

Reply:
That sounds like an excellent observation.

I think I'd agree to changing the definition of MAX_STREAM_ID to send the number of streams rather than the stream ID. We might also want to rename the frame to MAX_STREAM_COUNT.

There are certain aspects that are not elegant in the way MAX_STREAM_ID is defined now:
* we have two maximums; one for unidirectional streams
* we exchange counts using TP but we transmit maximum in the frame

These issues will be resolved by sending the counts.

Reply:
If we changed it to MAX_STREAM_COUNT, we'd have to use one additional bit from the type to indicate unidirectional/bidirectional, right? To me, my favorite thing (if we made that change) would be that I no longer have to validate if the peer is trying to set my limit on them (i.e. ownership bit).

Reply:
Assuming that MAX_STREAM_COUNT is supposed to encode the number of streams *since the beginning of the connection* (and not the concurrently open streams, as we had in some earlier version of the draft), I think this would be a good change to make, since we'd use the same encoding for the transport parameters and the frame.
Not sure if MAX_STREAM_COUNT is the best name for this, maybe we can go with something analogous to the transport parameters (initial_max_{uni,bidi}_streams), and name it MAX_STREAMS frame?